src/Analyzers/Core/Analyzers/MakeFieldReadonly/AbstractMakeFieldReadonlyDiagnosticAnalyzer.cs

```diff
@@ -235,11 +232,23 @@ private static bool IsFieldWrite(IFieldReferenceOperation fieldReference, ISymbo
                 if (instanceFieldWrittenInCtor || staticFieldWrittenInStaticCtor)
                 {
                     // Finally, ensure that the write is not inside a lambda or local function.
-                    if (fieldReference.TryGetContainingAnonymousFunctionOrLocalFunction() is null)
+                    if (fieldReference.TryGetContainingAnonymousFunctionOrLocalFunction() is not null)
+                        return true;
+
+                    // Legacy 'strict' C#/VB behavior.  If a special "feature:strict" (different from "option strict") flag Is on,
+                    // then this write Is only ok if the containing types are the same.  *Not* simply the original
+                    // definitions being the same (which the caller has already checked):
+                    //
+                    // https://github.com/dotnet/roslyn/blob/8770fb62a36157ed4ca38a16a0283d27321a01a7/src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs#L1201-L1203
+                    // https//github.com/dotnet/roslyn/blob/93d3aa1a2cf1790b1a0fe2d120f00987d50445c0/src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb#L1868-L1871
+                    if (fieldReference.SemanticModel?.SyntaxTree.Options.Features.ContainsKey("strict") is true &&
```

Comment:
❗ Let's _always_ follow the behavior that we should be getting if the compiler did not have a bug (i.e. don't check for `strict` here, but instead assume it is on).

Reply:
hrmm... i suppose that's an option.  this does seem esoteric enough that i doubt it would impact people.  will make that change.

Reply:
done.